autoload/cmake4vim.vim

```diff
@@ -158,6 +158,38 @@ function! cmake4vim#CMakeBuild(...) abort
     call utils#common#executeCommand(l:result, 0)
 endfunction
 
+" Builds current source
+function! cmake4vim#CompileCurrentSource() abort
+    if expand( '%:e' ) !=# 'cpp'
+        call utils#common#Warning( 'Current file is not a source file!' )
+        return
+    endif
+
+    let l:build_dir = utils#cmake#findBuildDir()
+    if empty( l:build_dir )
+        call utils#common#Warning( 'CMake project was not found!' )
+        return
+    endif
+
+    let l:cache_info = utils#cmake#cache#collectInfo( l:build_dir )
+    if empty( l:cache_info )
+        call utils#common#warning( 'CMake cache was not found!' )
+        return
+    endif
+
+    let l:generator = l:cache_info[ 'cmake' ][ 'generator' ]
+
+    " Select target
+    if l:generator ==# 'Unix Makefiles'
```

Comment:
I propose to use the same approach as for `utils#gen#common#getBuildCommand()`, `utils#gen#common#getTargets()` and etc. it allows to collect all generator specific code in one place.
Moreover for unsupported generators I propose to print warning and do nothing.